### PR TITLE
Remove argument length limiting

### PR DESCRIPTION
**Pull request by dqsully**
Arguments of the promisified functions shouldn't be limited to the number of arguments of the original function, most importantly because `util.promisify.custom` implementations may need more arguments than the functions they are wrapping to work effectively. For example, there is no promisified version of `https.get`, so I implemented my own, but it needs two arguments as opposed to one (one for the options to be passed through, another to return the request object through a callback):

```js
https.get[util.promisify.custom] = function (options, requestCallback) {
  let resolve, reject;
  let promise = new Promise((res, rej) => {
    resolve = res;
    reject = rej
  });

  try {
    let req = https.get(options, (res) => {
      resolve(res);
    });
    requestCallback(req);
  } catch(e) {
    reject(e);
  }

  return promise;
}
```

Also, I can't think of any reason to have that argument limiting, because if a function gets extra arguments, it just ignores them.

### CODE REVIEW

**Comment by doasync** [comment]
Yeah, it would work as expected with node functions. But check this example:

```js

const { promisify } = require('util');
const doAsync = require('doasync');

const myUtil = {
  upDelay (str, delay, cb) {
    setTimeout(() => {
      cb(str.toUpperCase());
    }, delay || 1000);
  }
};

const upDelayAsync = promisify(myUtil.upDelay);

// You have to pass `undefined` or cb will be the second param
upDelayAsync('foo', undefined).then(console.log, console.error);

// It just works as it should
doAsync(myUtil).upDelay('bar').then(console.log, console.error);

```
Your changes will break it. We just need an "improvement" (check the line I suggested above).

**Comment by dqsully** [comment]
**tl;dr** I don't think you should implement your fix, because you are (minutely) changing the way an API is used and don't have it documented. However, the decision is up to you, I would just recommend that if you keep your fix that you document it well. My programming won't change depending on which solution you use either way.

Personally, I think the job of accounting for optional variables should be on the API developer. For example:

```js
// To call upDelay, you need the undefined parameter anyway
myUtil.upDelay('foo', undefined, (err, val) => {/*code*/});

// So the api should stay the same, just with the callback removed
promisify(myUtil.upDelay)('foo', undefined).then(console.log, console.error)
```

Basically, while I think that your fix above is certainly handy, it is an extra feature added into a package meant to do one thing: wrap an object's functions with `util.promisify`. When you do more and change an API, especially without proper documentation, you can confuse your users. Also, your fix can create some rare issues, like with the example below.

```js
// I don't know why someone would put just three vars, but maybe it could help with documentation generators
function badAPI(var1, var2, var3, callback) {
  let a = Array.prototype.slice.call(arguments, 0);

  let cb = a.pop();

  // Sequentially goes through arguments and tries to call them as functions
  let runJob = () => {
    if(a.length == 0)
      callback();
    else
      a.shift()(runJob);
  }
  runJob();
}
doasync(badAPI)(() => console.log('hello'));
```

In this example, your script would fill `var2` and `var3` with `undefined`s, and because `badAPI` was written horribly, it would try to call those `undefined`s as functions and consequently throw errors and never call the callback. I don't know why someone would write a function like this, but the point is that it is possible, and in my opinion, a library like this meant to wrap anything should be able to handle any case in the way the developer expects - it should be transparent.

**Comment by dqsully** [comment]
You are also correct in the fact that my changes would be breaking, and would require updating to version `2.0.0` so that anyone including doasync v. `^1.1.3` (for example) will have to manually upgrade to `^2.0.0` in their `package.json` or through npm.

**Comment by doasync** [comment]
> a package meant to do one thing: wrap an object's functions with util.promisify

Ok, I agree with you on this. I really appreciate your help.
There will be version `2.0.0` with your pull request + some cool upgrade from me ;)

**Comment by dqsully** [comment]
Cool, thanks for letting me help and for making this awesome library available for everyone.
